integrations/DCMFloodlight/browser.js

```diff
@@ -0,0 +1,326 @@
+    this.googleNetworkId = config.googleNetworkId;
+    this.name = NAME;
+  }
+
+  /**
+   * Ref - https://support.google.com/campaignmanager/answer/7554821
+   */
+  init() {
+    logger.debug("===In init DCMFloodlight===");
+
+    const sourceUrl = `https://www.googletagmanager.com/gtag/js?id=DC-${this.advertiserId}`;
+    ScriptLoader("DCMFloodlight-integration", sourceUrl);
+
+    window.dataLayer = window.dataLayer || [];
+    window.gtag = function gtag() {
+      window.dataLayer.push(arguments);
+    };
+
+    // disable ad personalization
+    if (!this.allowAdPersonalizationSignals) {
+      window.gtag("set", "allow_ad_personalization_signals", false);
+    }
+
+    window.gtag("js", new Date());
+
+    if (this.conversionLinker) {
+      window.gtag("config", `DC-${this.advertiserId}`);
+    } else {
+      window.gtag("config", `DC-${this.advertiserId}`, {
+        conversion_linker: false,
+      });
+    }
+
+    // Google's cookie matching functionality
+    // Ref - https://developers.google.com/authorized-buyers/rtb/cookie-guide
+    if (this.doubleclickId && this.googleNetworkId) {
+      const image = document.createElement("img");
+      image.src = `https://cm.g.doubleclick.net/pixel?google_nid=${
+        this.googleNetworkId
+      }&google_hm=${btoa(this.analytics.anonymousId)}`;
+      document.getElementsByTagName("head")[0].appendChild(image);
+    }
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded DCMFloodlight===");
+    return window.dataLayer.push !== Array.prototype.push;
+  }
+
+  isReady() {
+    logger.debug("===In isReady DCMFloodlight===");
+    return window.dataLayer.push !== Array.prototype.push;
+  }
+
+  identify() {
+    logger.debug("[DCM Floodlight] identify:: method not supported");
+  }
+
+  track(rudderElement) {
+    logger.debug("===In DCMFloodlight track===");
+
+    const { message } = rudderElement;
+    let { event } = rudderElement.message;
+    let payload;
+    let customFloodlightVariable;
+    let salesTag;
+
+    const baseEndpoint = "https://ad.doubleclick.net/ddm/activity/";
+
+    if (!event) {
+      logger.error("[DCM Floodlight]:: event is required for track call");
+      return;
+    }
+
+    // Specifies how conversions will be counted for a Floodlight activity
+    let countingMethod = get(message, "properties.countingMethod");
+    if (!countingMethod) {
+      logger.error(
+        "[DCM Floodlight]:: countingMethod is required for track call"
+      );
+      return;
+    }
+    countingMethod = countingMethod.trim().toLowerCase().replace(/\s+/g, "_");
+
+    // find conversion event
+    // some() stops execution if at least one condition is passed and returns bool
+    event = event.trim().toLowerCase();
+    const conversionEventFound = this.conversionEvents.some(
+      (conversionEvent) => {
+        if (
+          conversionEvent &&
+          conversionEvent.eventName &&
+          conversionEvent.eventName.trim().toLowerCase() === event
+        ) {
```

Comment:
Is this correct to manipulate customer data? Here we are changing the event names to smaller cases. It may be possible customers may expect case-sensitive names.

Reply:
We are taking the event name from UI and validating it by bringing them to a common ground. We are not considering a case-sensitive name. `Order Complete` and `order complete` will be the same here.

Reply:
Yeah, that I understood. But I want to confirm, is it the right way to compare two event names?

Reply:
User can provide any name with respective to floodlight tag and those tag can have different meaning or actions. IMO doing this way we can avoid confusion for their actions. `Order Complete` for one task `oRdEr CoMpLeTe` for other and so on doesn't seem meaningful to have.

Reply:
Okay got it.